src/riak_kv_compile_tab.erl

```diff
@@ -151,7 +153,83 @@ get_ddl(BucketType, Version) when is_binary(BucketType), is_atom(Version) ->
             notfound
     end.
 
-%%
+%% Get the list of {TableName, Status} pairs, no matter what status they are in.
+-spec get_table_status_pairs() ->[{binary(), binary()}].
+get_table_status_pairs() ->
+    Matches = dets:match(?TABLE3, #row_v3{ table = '$1' }),
+    UTables = lists:usort(Matches),
+    Tables = [ {Table, get_table_status(Table)} ||
+               [Table|_T] <- UTables ],
+
+    %% sort by Status (thankfully in alphabetic sort order):
+    %% Active, Not Active, Undefined
+    SortByStatus = fun({L1,L2},{R1,R2}) -> {L2,L1} =< {R2,R1} end,
+    lists:sort(SortByStatus, Tables).
+
+get_table_status(Table) ->
+    case get_table_ddl(Table) of
+        {error, no_type} -> <<"Not Active">>;
+        {error, missing_helper_module} -> <<"Not Active">>;
+        {ok, _Mod, _DDL} -> get_table_status_by_version(Table);
+        _ -> get_table_status_by_version(Table)
+    end.
+
+get_table_status_by_version(Table) ->
+    Req = #tsqueryreq{},
```

Comment:
So I didn't have a problem with `check_table_feature_supported/2` returning a request.  It seems that part is actually used down in `riak_kv_ttb_ts:decode/2`.  My real issue was we are faking a request to essentially pass the table name down into `riak_kv_ts_util:is_ddl_version_supported/2`.  We are passing in `{ok, Req, {"perm", Table}}` which is pattern matched on `{ok, _, {_, Table}}` which just calls `is_ddl_version_supported(v2, Table)`.  I was just suggesting you expose that function to simplify the call and clarify the test condition (a version comparison.)

Less code == good.